File: CS/HorizontalMerging/Form1.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraGrid.Columns;

namespace HorizontalMerging
{
    public partial class Form1 : Form
    {
        MyGridViewHandler ViewHandler = null;
        public Form1()
        {
            InitializeComponent();
            ViewHandler = new MyGridViewHandler(gridView1);
        }

        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name1", typeof(string));
            tbl.Columns.Add("Name2", typeof(string));
            tbl.Columns.Add("Name3", typeof(string));
            tbl.Columns.Add("Name4", typeof(string));
            tbl.Columns.Add("Name5", typeof(string));
            for (int i = 0; i < RowCount; i++)
            {
                if (i == 1)
                    tbl.Rows.Add(new object[] { String.Format("Name{0}", i), "This is a long long string, which is merged for several columns", "", "", "" });
                else if (i == 3)
                    tbl.Rows.Add(new object[] { String.Format("Name{0}", i), "This is a long long string, which is merged for several columns", "", "", "Text" });
                else
                    tbl.Rows.Add(new object[] { String.Format("Name{0}", i), String.Format("Name{0}", i), String.Format("Name{0}", i), String.Format("Name{0}", i) });
            }
            return tbl;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            gridControl1.DataSource = CreateTable(20);
            gridView1.Columns[0].Fixed = FixedStyle.Left;
            gridView1.Columns[4].Width = 300;
            gridControl1.ForceInitialize();
            ViewHandler.MergeCells(gridView1.GetRowCellValue(1, "Name2").ToString(), gridView1.GetDataSourceRowIndex(1), new GridColumn[] { gridView1.Columns[2], gridView1.Columns[3], gridView1.Columns[4] });
            ViewHandler.MergeCells(gridView1.GetRowCellValue(3, "Name2").ToString(), gridView1.GetDataSourceRowIndex(3), new GridColumn[] { gridView1.Columns[2], gridView1.Columns[3] });
        }
    }
 }

File: VB/HorizontalMerging/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Columns

Namespace HorizontalMerging
	Partial Public Class Form1
		Inherits Form

		Private ViewHandler As MyGridViewHandler = Nothing
		Public Sub New()
			InitializeComponent()
			ViewHandler = New MyGridViewHandler(gridView1)
		End Sub

		Private Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Name1", GetType(String))
			tbl.Columns.Add("Name2", GetType(String))
			tbl.Columns.Add("Name3", GetType(String))
			tbl.Columns.Add("Name4", GetType(String))
			tbl.Columns.Add("Name5", GetType(String))
			For i As Integer = 0 To RowCount - 1
				If i = 1 Then
					tbl.Rows.Add(New Object() { String.Format("Name{0}", i), "This is a long long string, which is merged for several columns", "", "", "" })
				ElseIf i = 3 Then
					tbl.Rows.Add(New Object() { String.Format("Name{0}", i), "This is a long long string, which is merged for several columns", "", "", "Text" })
				Else
					tbl.Rows.Add(New Object() { String.Format("Name{0}", i), String.Format("Name{0}", i), String.Format("Name{0}", i), String.Format("Name{0}", i) })
				End If
			Next i
			Return tbl
		End Function

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			gridControl1.DataSource = CreateTable(20)
			gridView1.Columns(0).Fixed = FixedStyle.Left
			gridView1.Columns(4).Width = 300
			gridControl1.ForceInitialize()
			ViewHandler.MergeCells(gridView1.GetRowCellValue(1, "Name2").ToString(), gridView1.GetDataSourceRowIndex(1), New GridColumn() { gridView1.Columns(2), gridView1.Columns(3), gridView1.Columns(4) })
			ViewHandler.MergeCells(gridView1.GetRowCellValue(3, "Name2").ToString(), gridView1.GetDataSourceRowIndex(3), New GridColumn() { gridView1.Columns(2), gridView1.Columns(3) })
		End Sub
	End Class
End Namespace
